File: Orimath/IO/GlobalSetting.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Orimath.IO
{
    public sealed class GlobalSetting : ICloneable
    {
        [Display(Name = "表示サイズ")]
        [Range(1, 5000)]
        public int ViewSize { get; set; } = 512;

        [Editable(false)]
        public double Height { get; set; } = 600.0;

        [Editable(false)]
        public double Width { get; set; } = 800.0;

        [Editable(false)]
        public double Left { get; set; } = double.NaN;

        [Editable(false)]
        public double Top { get; set; } = double.NaN;

        public object Clone() => MemberwiseClone();
    }
}

File: Orimath/IO/GlobalSetting.fs
namespace Orimath.IO
open System
open System.ComponentModel.DataAnnotations

[<ReferenceEquality; NoComparison>]
type GlobalSetting =
    {
        [<Display(Name = "{PaperSize}Paper size")>]
        [<Range(1, 5000)>]
        mutable ViewSize: int
        [<Editable(false)>]
        mutable Height: float
        [<Editable(false)>]
        mutable Width: float
    }
    with

    static member CreateDefault() = { ViewSize = 512; Height = 600.0; Width = 800.0 }

    member this.Clone() = this.MemberwiseClone() :?> GlobalSetting

    interface ICloneable with
        member this.Clone() = this.MemberwiseClone()
